src/nbl/video/CVKLogicalDevice.h

```diff
@@ -124,51 +327,472 @@ class CVKLogicalDevice final : public ILogicalDevice
+
+                        VkBufferView vk_bufferView = static_cast<const CVulkanBufferView*>(pDescriptorWrites[i].info[j].desc.get())->getInternalObject();
+                        vk_bufferViews[j] = vk_bufferView;
+                    }
+
+                    vk_writeDescriptorSets[i].pTexelBufferView = vk_bufferViews + bufferViewOffset;
+                    bufferViewOffset += pDescriptorWrites[i].count;
+                } break;
+            }
+        }
+
+        assert(descriptorCopyCount <= MAX_DESCRIPTOR_COPY_COUNT);
+        VkCopyDescriptorSet vk_copyDescriptorSets[MAX_DESCRIPTOR_COPY_COUNT];
+
+        for (uint32_t i = 0u; i < descriptorCopyCount; ++i)
+        {
+            vk_copyDescriptorSets[i].sType = VK_STRUCTURE_TYPE_COPY_DESCRIPTOR_SET;
+            vk_copyDescriptorSets[i].pNext = nullptr; // pNext must be NULL
+
+            // if (pDescriptorCopies[i].srcSet->getAPIType() != EAT_VULKAN)
+            //     continue;
+            vk_copyDescriptorSets[i].srcSet = static_cast<const CVulkanDescriptorSet*>(pDescriptorCopies[i].srcSet)->getInternalObject();
+
+            vk_copyDescriptorSets[i].srcBinding = pDescriptorCopies[i].srcBinding;
+            vk_copyDescriptorSets[i].srcArrayElement = pDescriptorCopies[i].srcArrayElement;
+
+            // if (pDescriptorCopies[i].dstSet->getAPIType() != EAT_VULKAN)
+            //     continue;
+            vk_copyDescriptorSets[i].dstSet = static_cast<const CVulkanDescriptorSet*>(pDescriptorCopies[i].dstSet)->getInternalObject();
+
+            vk_copyDescriptorSets[i].dstBinding = pDescriptorCopies[i].dstBinding;
+            vk_copyDescriptorSets[i].dstArrayElement = pDescriptorCopies[i].dstArrayElement;
+            vk_copyDescriptorSets[i].descriptorCount = pDescriptorCopies[i].count;
+        }
+
+        vkUpdateDescriptorSets(m_vkdev, descriptorWriteCount, vk_writeDescriptorSets, descriptorCopyCount, vk_copyDescriptorSets);
+    }
+
+    core::smart_refctd_ptr<IDriverMemoryAllocation> allocateDeviceLocalMemory(
+        const IDriverMemoryBacked::SDriverMemoryRequirements& additionalReqs) override
+    {
+        // Todo(achal): Remove this, and integrate it with
+        // IDriverMemoryBacked::SDriverMemoryRequirements if possible. Probably we can cache this
+        // in a struct of 5 uint32_t's (deviceLocal, spillover, upstreaming,
+        // downstreaming, cpuSizeGPUVisible) per CVulkanPhysicalDevice and just access
+        // the relevant index in here.
+#if 0
+        uint32_t memoryTypeIndex = ~0u;
+        {
+            if (m_physicalDevice->getAPIType() != EAT_VULKAN)
+                return nullptr;
+
+            VkPhysicalDevice vk_physicalDevice = static_cast<const CVulkanPhysicalDevice*>(m_physicalDevice.get())->getInternalObject();
+
+            VkPhysicalDeviceMemoryProperties vk_physicalDeviceMemoryProperties;
+            vkGetPhysicalDeviceMemoryProperties(vk_physicalDevice, &vk_physicalDeviceMemoryProperties);
```

Comment:
This you can cache in VulkanPhysicalDevice

Reply:
resolved